src/Std/Data/DHashMap/Internal/List/Associative.lean

```diff
@@ -16,6 +17,22 @@ the contents of this file.
 File contents: Verification of associative lists
 -/
 
+
+-- TODO: should this go somwhere else?
+theorem List.find?_isSome_of_map_fst_contains {α : Type u} {β : α -> Type v} [BEq α] {l : List ((a : α) × β a)} {k : α} :
+    (l.map Sigma.fst).contains k ->
+      ((l.map Sigma.fst).reverse.find? (fun a => k == a)).isSome := by
```

Comment:
Do you really care about `.find?.isSome`? Presumably there is a simp lemma that reduces this to `List.any`.